openmc/deplete/batchwise.py

```diff
@@ -0,0 +1,1157 @@
+        Default to 'constant-volume'
+    bracketed_method : {'brentq', 'brenth', 'ridder', 'bisect'}, optional
+        Solution method to use.
+        This is equivalent to the `bracket_method` parameter of the
+        `search_for_keff`.
+        Defaults to 'brentq'.
+    tol : float
+        Tolerance for search_for_keff method.
+        This is equivalent to the `tol` parameter of the `search_for_keff`.
+        Default to 0.01
+    target : Real, optional
+        This is equivalent to the `target` parameter of the `search_for_keff`.
+        Default to 1.0.
+    print_iterations : Bool, Optional
+        Whether or not to print `search_for_keff` iterations.
+        Default to True
+    search_for_keff_output : Bool, Optional
+        Whether or not to print transport iterations during  `search_for_keff`.
+        Default to False
+    Attributes
+    ----------
+    burn_mats : list of str
+        List of burnable materials ids
+    local_mats : list of str
+        All burnable material IDs being managed by a single process
+        List of burnable materials ids
+    """
+    def __init__(self, operator, model, bracket, bracket_limit,
+                 density_treatment = 'constant-volume', bracketed_method='brentq',
+                 tol=0.01, target=1.0, print_iterations=True,
+                 search_for_keff_output=True):
+
+        self.operator = operator
+        self.burn_mats = operator.burnable_mats
+        self.local_mats = operator.local_mats
+        self.model = model
+        self.geometry = model.geometry
+
+        check_value('density_treatment', density_treatment,
+                                        ('constant-density','constant-volume'))
+        self.density_treatment = density_treatment
+
+        check_iterable_type('bracket', bracket, Real)
+        check_length('bracket', bracket, 2)
+        check_less_than('bracket values', bracket[0], bracket[1])
+        self.bracket = bracket
+
+        check_iterable_type('bracket_limit', bracket_limit, Real)
+        check_length('bracket_limit', bracket_limit, 2)
+        check_less_than('bracket limit values',
+                         bracket_limit[0], bracket_limit[1])
+        self.bracket_limit = bracket_limit
+
+        check_value('bracketed_method', bracketed_method,
+                   _SCALAR_BRACKETED_METHODS)
+        self.bracketed_method = bracketed_method
+
+        check_type('tol', tol, Real)
+        self.tol = tol
+
+        check_type('target', target, Real)
+        self.target = target
+
+        self.print_iterations = print_iterations
+        self.search_for_keff_output = search_for_keff_output
+
+    @property
+    def bracketed_method(self):
+        return self._bracketed_method
+
+    @bracketed_method.setter
+    def bracketed_method(self, value):
+        check_value('bracketed_method', value, _SCALAR_BRACKETED_METHODS)
+        if value != 'brentq':
+            warn('WARNING: brentq bracketed method is recommended')
```

Comment:
Can we instead put this in some documentation somewhere? If other methods are not recommended, why do we support them? Sure it's good to have more options, and there is usually a good case to be made for testing out new algorithms. 

I feel a better user experience would be to have some documentation indicating the default `brentq` method is recommended and this warning be removed.